agent/agent.go

```diff
@@ -2785,6 +2807,56 @@ func (a *Agent) unloadChecks() error {
 // loadProxies will load connect proxy definitions from configuration and
 // persisted definitions on disk, and load them into the local agent.
 func (a *Agent) loadProxies(conf *config.RuntimeConfig) error {
+	persistedProxies := make(map[string]persistedProxy)
+	a.proxyLock.Lock()
+	defer a.proxyLock.Unlock()
+
+	// Load any persisted proxies
+	proxyDir := filepath.Join(a.config.DataDir, proxyDir)
+	files, err := ioutil.ReadDir(proxyDir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("Failed reading proxies dir %q: %s", proxyDir, err)
+		}
```

Comment:
I forgot for a moment you can do a range on a nil slice and it will do nothing. Otherwise the else would be necessary to prevent that as we wouldn't return for an IsNotExist error.